core/validatorapi/validatorapi.go

```diff
@@ -56,7 +56,7 @@ func NewComponentInsecure(_ *testing.T, eth2Cl eth2wrap.Client, shareIdx int) (*
 
 // NewComponent returns a new instance of the validator API core workflow component.
 func NewComponent(eth2Cl eth2wrap.Client, pubShareByKey map[*bls_sig.PublicKey]*bls_sig.PublicKey,
```

Comment:
you are passing in the same data `pubShareByKey` twice but in different formats

Reply:
That's right. Maybe we can replace `pubShareByKey` with `allPubSharesByKey`? Since we can calculate relevant pubkey mappings from the latter

Reply:
yeah, lets try that